File: app/src/main/java/com/chernybro/loftcoin/ui/activity/wallets/WalletsAdapter.java
package com.chernybro.loftcoin.ui.activity.wallets;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chernybro.loftcoin.databinding.IWalletBinding;


class WalletsAdapter extends RecyclerView.Adapter<WalletsAdapter.ViewHolder> {

    private LayoutInflater inflater;

    @Override
    public int getItemCount() {
        return 10;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(IWalletBinding.inflate(inflater, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

    }

    @Override
    public void onAttachedToRecyclerView(@NonNull RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
        inflater = LayoutInflater.from(recyclerView.getContext());
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        ViewHolder(@NonNull IWalletBinding binding) {
            super(binding.getRoot());
        }

    }

}

File: app/src/main/java/com/chernybro/loftcoin/ui/activity/wallets/WalletsAdapter.kt
package com.chernybro.loftcoin.ui.activity.wallets

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.chernybro.loftcoin.databinding.IWalletBinding

internal class WalletsAdapter : RecyclerView.Adapter<WalletsAdapter.ViewHolder>() {
    private var inflater: LayoutInflater? = null
    override fun getItemCount(): Int {
        return 10
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            IWalletBinding.inflate(
                inflater!!, parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {}
    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
        super.onAttachedToRecyclerView(recyclerView)
        inflater = LayoutInflater.from(recyclerView.context)
    }

    internal class ViewHolder(binding: IWalletBinding) : RecyclerView.ViewHolder(binding.root)
}